domains/certificates/Query.API/API/Query.API/Projections/CertificatesByOwnerProjection.cs

```diff
@@ -1,119 +1,153 @@
+    {
+        var source = operations.Load<CertificatesByOwnerView>(productionCertificateTransferred.Source)
+                     ?? throw new CertificateDomainException(
+                         productionCertificateTransferred.CertificateId,
+                         $"Cannot transfer from {productionCertificateTransferred.Source}. {productionCertificateTransferred.CertificateId} cannot be found"
+                     );
+        var cert = source.Certificates[productionCertificateTransferred.CertificateId];
 
-    public Dictionary<Guid, CertificateView> Certificates { get; set; } = new();
+        RemoveCertificateFromSource(operations, source, productionCertificateTransferred);
+        return AddCertificateToTarget(operations, productionCertificateTransferred, cert);
+    }
 
-    public CertificateList ToApiModel()
+    private static CertificatesByOwnerView AddCertificateToTarget(IDocumentOperations operations,
+        ProductionCertificateTransferred productionCertificateTransferred, CertificateView certificateView)
     {
-        var certificates = Certificates.Values
-            .Select(c => new Certificate
-            {
-                Id = c.CertificateId,
-                GSRN = c.GSRN,
-                GridArea = c.GridArea,
-                DateFrom = c.DateFrom,
-                DateTo = c.DateTo,
-                Quantity = c.Quantity,
-                TechCode = c.TechCode,
-                FuelCode = c.FuelCode
-            });
-
-        return new CertificateList
+        var view = operations.Load<CertificatesByOwnerView>(productionCertificateTransferred.Target);
+
+        if (view == null)
         {
-            Result = certificates
-                .OrderByDescending(c => c.DateFrom)
-                .ThenBy(c => c.GSRN)
-                .ToArray()
-        };
+            return new CertificatesByOwnerView
+            {
+                Owner = productionCertificateTransferred.Target,
+                Certificates =
+                {
+                    [certificateView.CertificateId] = certificateView
+                }
+            };
+        }
+
+        view.Certificates.Add(certificateView.CertificateId, certificateView);
+        return view;
     }
-}
 
-public class CertificateView
-{
-    public Guid CertificateId { get; set; }
-    public long DateFrom { get; set; }
-    public long DateTo { get; set; }
-    public long Quantity { get; set; }
-    public string GSRN { get; set; } = "";
-    public string GridArea { get; set; } = "";
-    public string TechCode { get; set; } = "";
-    public string FuelCode { get; set; } = "";
-    public CertificateStatus Status { get; set; }
-}
+    private static void RemoveCertificateFromSource(IDocumentOperations operations, CertificatesByOwnerView source,
+        ProductionCertificateTransferred productionCertificateTransferred)
+    {
+        source.Certificates.Remove(productionCertificateTransferred.CertificateId);
+        operations.Update(source);
```

Comment:
It seems like "the pattern" is broken here. It appears that there should only be one update/store on line 45 with `operations.Store(view);`, but here is an extra update of a view (`source`) and is a bit hidden.

I am not sure if I have an easy fix other than going with a different approach. A different approach could be to start with an empty list of views and then let the methods for applying the events add to that list; in the end, it is just looping through that list and calling `opertions.Store(item)`